# Семинар 7 ДЗ

## Задача 1. 
Даны две независимые выборки. Не соблюдается условие нормальности

$x1 = 380, 420, 290$<br>
$y1 = 140, 360, 200, 900$

Сделайте вывод по результатам, полученным с помощью функции, имеются ли статистические различия между группами?

## Решение

Для решения данной задачи мы будем использовать Критерий Манна – Уитни

Его следует использовать для сравнения двух независимых выборок, если не соблюдается условие нормальности, или, если выборки очень маленького объема.

In [17]:
from scipy import stats
import numpy as np

In [18]:
x = np.array([380, 420, 290])
y = np.array([140, 360, 200, 900])

In [19]:
stats.mannwhitneyu(x, y)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

Как мы можем интерпретировать полученные данные?

$pvalue \approx 0.63$

$pvalue$ больше любых стандартных $\alpha$, следовательно нулевая гипотеза об отсутствии статистически значимых различий принимается.

Статистических различий между группами нет!


## Задача 2. 
Исследовалось влияние препарата на уровень давления пациентов. 

Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. 

Есть ли статистически значимые различия между измерениями давления? 

В выборках не соблюдается условие нормальности.

$1$е измерение до приема препарата: $150, 160, 165, 145, 155$<br>
$2$е измерение через $10$ минут: $140, 155, 150, 130, 135$<br>
$3$е измерение через $30$ минут: $130, 130, 120, 130, 125$<br>

## Решение

Для данного случая применим критерий Фридмана, так как в данном случае мы имеем множественные, повторные измерения.

In [25]:
A = np.array ([150, 160, 165, 145, 155])
B = np.array ([140, 155, 150, 130, 135])
C = np.array ([130, 130, 120, 130, 125])

In [26]:
stats.friedmanchisquare(A, B, C)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

Как мы можем интерпретировать полученный результат?

$pvalue \approx 0.0083$

$0.0083 < 0.05$ 

$pvalue$ существенно ниже среднего уровня значимости $\alpha = 0.05$, следовательно нулевая гипотеза об отсутствии статистически значимых различий отвергается, статистически значимые различия между измерениями давления есть!


## Задача 3. 
Сравните $1$ и $2$-е измерения, предполагая, что $3$го измерения через $30$ минут не было. 

Есть ли статистически значимые различия между измерениями давления?

## Решение

Так как в данном случае мы имеем две зависимые выборки, то мы применим критерий Уилкоксона

In [27]:
A = np.array ([150, 160, 165, 145, 155])
B = np.array ([140, 155, 150, 130, 135])
C = np.array ([130, 130, 120, 130, 125])
A-B

array([10,  5, 15, 15, 20])

In [33]:
stats.wilcoxon(A, B)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

$pvalue > \alpha =>$

Нулевая гипотеза о равенстве не отвергается, таким образом мы можем сделать вывод о том, что через десять минут прием препарата не оказывает статистически значимого влияния.

В качестве эксперимента, в дополнение к заданию, сравним группы $A$ и $C$

In [24]:
stats.wilcoxon(A, C)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

Мы видим тот же результат, так как статистика в обоих случаях равна $0$, в обоих последующих измерениях давление снизилось. 

## Задача 4. 

Даны $3$ группы учеников плавания. Не соблюдается условие нормальности.

В $1$ группе время на дистанцию $50$ м составляют:

Первая группа: $56, 60, 62, 55, 71, 67, 59, 58, 64, 67$<br>
Вторая группа: $57, 58, 69, 48, 72, 70, 68, 71, 50, 53$<br>
Третья группа: $57, 67, 49, 48, 47, 55, 66, 51, 54$<br>

Есть ли статистически значимые различия между группами?

## Решение

Здесь мы имеем три независимые выборки, поэтому применим критерий Критерий Крускала – Уоллиса

In [34]:
gr1 =([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
gr2 = ([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
gr3 = ([57, 67, 49, 48, 47, 55, 66, 51, 54])

stats.kruskal(gr1, gr2, gr3)

KruskalResult(statistic=5.465564058257224, pvalue=0.06503809985904942)

Так как $pvalue \approx 0.065$ , что больше $\alpha = 0.05$ мы можем сделать вывод, что статистически занчимых различий между группами нет!

## Задача 5. 
Заявляется, что партия изготавливается со средним арифметическим $2,5$ см. 

Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. 

Объем выборки $10$,<br> 
уровень статистической значимости $5\%$

$2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34$

## Решение

1. Сформулируем нулевую и альтернативную гипотезу<br>
$H_0: \mu_x = 2.5$<br>
$H_0: \mu_x \neq 2.5$

2. Уровень статистической значимости $5\%$, $\alpha = 0.05$

3. Так как в этой задаче мы не знаем СКО генеральной совокупности, то мы будем использовать t-статистику, которая имеет распределение Стьюдента.<br>
 <br>
$t_{наблюдаемое} = \dfrac{\overline{X_1}−\mu}{\dfrac{\sigma_x}{\sqrt{𝑛}}}$

Рассчитаем необходимые данные при помощи python

In [36]:
alpha = 0.05
n = 10
mu = 2.5

x = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])

In [37]:
sigma = x.std(ddof=1)
x_mean = x.mean()
sigma, x_mean

(0.1572542173961923, 2.5279999999999996)

4. Определим критические области:<br>

In [38]:
t1 = stats.t.ppf(alpha/2, df=n-1)
t2 = stats.t.ppf(1 - alpha/2, df=n-1)

t1, t2

(-2.262157162740992, 2.2621571627409915)

5. Расчитаем $t_{наблюдаемое}$

In [39]:
t_emp = (x_mean - mu)/ (sigma / n ** 0.5)
t_emp

0.5630613661802959

$t_{наблюдаемое}$ не принадлежит критической области:

$-2.26 \leq 0.56 \leq 2.26 $, 

следовательно гипотеза $H_0$ не отвергается!